src/core/include/math/hal/intnat/transformnat-impl.h

```diff
@@ -551,6 +549,30 @@ void NumberTheoreticTransformNat<VecType>::InverseTransformFromBitReverseInPlace
             }
         }
     }
+    const uint32_t j2 = n >> 1;
+    for (uint32_t j1{0}; j1 < j2; ++j1) {
+        auto loVal{(*element)[j1]};
+        auto hiVal{(*element)[j1 + j2]};
+#if defined(__GNUC__) && !defined(__clang__)
+        auto omegaFactor{loVal};
+        if (omegaFactor < hiVal)
+            omegaFactor += modulus;
+        omegaFactor -= hiVal;
+        loVal += hiVal;
+        if (loVal >= modulus)
+            loVal -= modulus;
+        omegaFactor.ModMulFastConstEq(omega1Inv, modulus, preconOmega1Inv);
+        (*element)[j1 + 0]  = loVal;
+        (*element)[j1 + j2] = omegaFactor;
+#else
+        (*element)[j1] += hiVal - (hiVal >= (modulus - loVal) ? modulus : 0);
+        auto omegaFactor = loVal + (hiVal > loVal ? modulus : 0) - hiVal;
+        omegaFactor.ModMulFastConstEq(omega1Inv, modulus, preconOmega1Inv);
+        (*element)[j1 + j2] = omegaFactor;
+#endif
+    }
+    for (uint32_t i = 0; i < j2; ++i)
```

Comment:
Why do we need more than 1 multiply with N inverse?
The optimization I proposed should absorb all the multiply by N inverse, and we only need to multiply (*element)[0] by N inverse.

Reply:
The optimization you originally proposed, which incorporated (N-1) inverse multiplies incorporated into the twiddles, would require ~2x twiddle memory, wouldn't be backwards compatible, and would require extra if statements to select appropriate twiddle data making the optimized transform run slower than the original. This code only incorporates the inverse into bitreversed(omega1), which is multiplied by the last N/2 elements in the final intt stage. This loop is added to multiply the remaining N/2 elements by the inverse. I would have added this multiply into the final intt stage as well, but benchmarking showed it was faster to just have an extra loop at the end.